apps/maestro/src/lib/hooks/useAccount.ts

```diff
@@ -30,54 +30,11 @@ export function useAccount(): CombinedAccountInfo {
     [wagmiAccount?.chain, evmChains]
   );
 
-  // TODO: do it more generic
   return {
     address: wagmiAccount.address || (mystenAccount?.address as `0x${string}`),
     isConnected: isWagmiConnected || isMystenConnected,
     isDisconnected: !isWagmiConnected && !isMystenConnected,
-    chain:
-      wagmiAccount.chain ||
-      (isMystenConnected
-        ? NEXT_PUBLIC_NETWORK_ENV === "mainnet"
-          ? {
-              id: 101,
-              name: "Sui",
-              nativeCurrency: {
-                name: "SUI",
-                symbol: "SUI",
-                decimals: 9,
-              },
-              rpcUrls: {
-                default: { http: ["https://sui-rpc.publicnode.com"] },
-                public: { http: ["https://sui-rpc.publicnode.com"] },
-              },
-              blockExplorers: {
-                default: {
-                  name: "Sui Explorer",
-                  url: "https://suiexplorer.com",
-                },
-              },
-            }
-          : {
-              id: 103,
-              name: "Sui Testnet",
-              nativeCurrency: {
-                name: "SUI",
-                symbol: "SUI",
-                decimals: 9,
-              },
-              rpcUrls: {
-                default: { http: ["https://fullnode.testnet.sui.io:443"] },
-                public: { http: ["https://fullnode.testnet.sui.io:443"] },
-              },
-              blockExplorers: {
-                default: {
-                  name: "Sui Explorer",
-                  url: "https://suiscan.xyz/testnet",
-                },
-              },
-            }
-        : undefined),
+    chain: wagmiAccount.chain || suiChainConfig,
```

Comment:
Be prepared for other chains supported here.
Don't fallback to `suiChainConfig` without any reasonable condition. Probably also check for `isMystenConnected` as well?